File: parciales/ED1-Parcial1-2021-1--002/codigo/WaitingLine.py
import numpy as np
class WaitingLine:

  def __init__(self, n):
    self.e = np.zeros(n)
    self.index = 0;
    self.started = False;

  def isEmpty(self): 
    return (self.index == 0 and not self.started) or self.index <= 0

  def enqueue(self,x):
    if self.index == self.e.size:
      raise Exception("Full")
    self.started = True
    self.e[self.index] = x
    self.index = self.index + 1

  def dequeue(self):
    if self.isEmpty():
      raise Exception("Empty")
    x = self.e[0]
    for i in range (0, self.index - 1):
      self.e[i] = self.e[i + 1]
    self.index = self.index - 1
    return x

def __main__():
    w = WaitingLine(5)
    w.enqueue(2)
    w.enqueue(3) 
    print(w.dequeue())
    print(w.dequeue())
    print(w.dequeue())

__main__()

File: parciales/ED1-Parcial1-2021-1--002/codigo/WaitingLine.java
class WaitingLine {
  int e[];
  int index;
  boolean started;

  WaitingLine(int n) {
    e = new int[n];
    index = 0;
    started = false;
  }

  public boolean isEmpty() {
    return (index == 0 && !started) | index <= 0;
  }

  public void enqueue(int x){
    if (index == e.length) {
      throw new RuntimeException("Full");
    }
    started = true;
    e[index] = x;
    index++;
  }

  public int dequeue(){
    if (isEmpty())
      throw new RuntimeException("Empty");
    int x = e[0];
    for (int i = 0; i < index - 1; i++) {
      e[i] = e[i + 1];
    }
    index--;
    return x;
  }

  public static void main(String[] args){
    WaitingLine w = new WaitingLine(5);
    w.enqueue(2);
    w.enqueue(3);
    System.out.println(w.dequeue());
    System.out.println(w.dequeue());
    System.out.println(w.dequeue());
    System.out.println(w.dequeue());
  }
}
